resources/dw/dataviz/resource_dataviz.go

```diff
@@ -0,0 +1,371 @@
+// Copyright 2025 Cloudera. All Rights Reserved.
+//
+// This file is licensed under the Apache License Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, either express or implied. Refer to the License for the specific
+// permissions and limitations governing your use of the file.
+
+package dataviz
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/cdp"
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/gen/dw/client/operations"
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/gen/dw/models"
+	"github.com/cloudera/terraform-provider-cdp/utils"
+)
+
+type datavizResource struct {
+	client *cdp.Client
+}
+
+var (
+	_ resource.Resource                   = (*datavizResource)(nil)
+	_ resource.ResourceWithConfigure      = (*datavizResource)(nil)
+	_ resource.ResourceWithImportState    = (*datavizResource)(nil)
+	_ resource.ResourceWithValidateConfig = (*datavizResource)(nil)
+)
+
+const (
+	resourceTemplateDefault string = "viz-default"
+	resourceTemplateLow     string = "viz-low"
+	resourceTemplateMedium  string = "viz-medium"
+	resourceTemplateLarge   string = "viz-large"
+)
+
+func NewDataVizResource() resource.Resource {
+	return &datavizResource{}
+}
+
+func (r *datavizResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	r.client = utils.GetCdpClientForResource(req, resp)
+}
+
+func (r *datavizResource) ValidateConfig(ctx context.Context, req resource.ValidateConfigRequest, resp *resource.ValidateConfigResponse) {
+	var data resourceModel
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if admGrp := data.AdminGroups; len(admGrp.Elements()) == 0 {
+		resp.Diagnostics.AddAttributeError(
+			path.Root("admin_groups"),
+			"Invalid administrator groups",
+			"The admin_groups must have at least one group.",
+		)
+	}
+
+	// If resource template not defined then default will be applied -> viz-default, otherwise we validate
+	if !data.ResourceTemplate.IsNull() && data.ResourceTemplate.ValueString() != "" {
```

Comment:
Nit: For readability , maybe you can use this and add all conditions to a single line

        resourceTemplatePossibleValues := []string{
		        resourceTemplateDefault,
		        resourceTemplateLow,
		        resourceTemplateMedium,
		        resourceTemplateLarge,
	        }
        


	    if !data.ResourceTemplate.IsNull() && !isAnyOf(data.ResourceTemplate.ValueString(), resourceTemplatePossibleValues...) {
    
		    resp.Diagnostics.AddAttributeError(
			    path.Root("resource_template"),

Reply:
Your suggestion is not the same as the current code. In the current version we also handle the `""` (empty string case). To make it a single liner we'd have to add the `""` to the `resourceTemplatePossibleValues` which was making the error message a bit ugly saying an empty string is a chosen value..

Reply:
I keep this code unchanged. Thanks for the suggestion.